File: solutions/1493.py
from typing import List

class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        result = 0
        pos_0 = -1
        l, r = 0, 0
        while r < len(nums):
            if nums[r] != 1:
                if pos_0 == -1:
                    pos_0 = r
                else:
                    result = max(result, r - l - 1)
                    l = pos_0 + 1
                    pos_0 = r
            r += 1
        result = max(result, r - l - 1)

        return result


nums = [0, 0]
solution = Solution()
print(solution.longestSubarray(nums))

File: solutions/1493.go
package main


func longestSubarray(nums []int) int {
    result := 0
    idx0, l, r := -1, 0, 0

    for r < len(nums) {
        if nums[r] == 0 {
            if idx0 != -1 {
                result = max(result, r - l - 1)
                l = idx0 + 1
            }
            idx0 = r
        }
        r++
    }
    result = max(result, r - l - 1)
    return result
}